### ISSUE TITLE
solve_pnp_dlt: a zero weight still enters the normalization and the degeneracy check, so planar points pass with a wrong pose

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4799]
`solve_pnp_dlt`'s `weights` scales the two DLT rows that each point contributes, so a weight of `0` removes the point from the linear system. The point still takes part in two other steps:

- the normalization of both point sets (their mean and scale);
- the check that `world_points` is not degenerate (the `svd_eps` singular-value test).

So a zero weight does not drop the point, although the docstring said it did. The degeneracy check is where this shows. Ten coplanar points plus two off-plane points solve correctly. Give the two off-plane points weight `0` and the system is rank-deficient, which is the coplanar case the check exists to reject. The check still passes, because it looks at the unweighted points, and the call returns a wrong pose:

```python
import torch
from kornia.geometry.calibration import solve_pnp_dlt

torch.manual_seed(0)
dt = torch.float64
K = torch.tensor([[[500.0, 0, 320], [0, 500, 240], [0, 0, 1]]], dtype=dt)
xy = torch.rand(10, 2, dtype=dt) * 4 - 2
planar = torch.cat([xy, torch.full((10, 1), 8.0, dtype=dt)], 1)[None]  # plane z = 8, true pose [I | 0]
points = torch.cat([planar, torch.tensor([[[0.0, 0.0, 12.0], [1.0, -1.0, 15.0]]], dtype=dt)], 1)

def proj(w):
    p = (K[0] @ w[0].T).T
    return (p[:, :2] / p[:, 2:])[None]

w = torch.ones(1, 12, dtype=dt)
print(solve_pnp_dlt(points, proj(points), K, weights=w)[0, :, 3])  # ~0: correct
w[0, -2:] = 0
print(solve_pnp_dlt(points, proj(points), K, weights=w)[0, :, 3])  # (-181.5, 4.8, -135.0), no raise
try:
    solve_pnp_dlt(planar, proj(planar), K)
except AssertionError:
    print("the ten coplanar points alone: AssertionError")        # what the zero weights should give
```

On noisy data the zero-weight point also moves the answer through the normalization. With ten noisy points plus one far point at weight `0`, the result differs from the ten-point solve by `3e-3` to `1.6e-2` in `[R | t]`. With exact data the two agree to rounding.

A fix would compute the normalization and the degeneracy check from the weighted points, or from the points with nonzero weight. After that, a zero weight would be equivalent to removing the point. #4790 documents this as a wart on `solve_pnp_dlt` and pins it with the planar case: `test_wart_zero_weight_point_still_enters_the_degeneracy_check_4799`.

kornia `main` @ 57cd68e6f, torch 2.14.0, CPU.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
